Lesson31/Kadir/my-app/src/components/AddButton/index.jsx

```diff
@@ -0,0 +1,43 @@
+import React, { useState } from "react";
+import "./styles.css";
+
+export const AddButton = ({ key, data, setData}) => {
+  const [text, setText] = useState("");
+  const [newItem, setNewItem] = useState(); // Define a state to store the newItem
+
+  const handleChange = (event) => {
+    setText(event.target.value);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    text.trim() !== "" &&
+      setNewItem({
+        key: key + 1,
+        id: Math.floor(Math.random() * 1000),
+        title: text,
+        completed: false,
+      });
+    setText(""); // Clear the input field
+  };
+
+  const handleAddItem = (newItem) => {
+    setData([...data, newItem]);
+  };
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <input
+        type="text"
+        className="input-field"
+        placeholder="Add new list item"
+        value={text}
+        onChange={handleChange}
+
+      />
+      <button type="submit" className="btn-submit" onClick={handleAddItem}>
```

Comment:
You have an `onSubmit` event listener added to the form element, no need to add another `onClick` handler to the submit button as the click event will be handled anyway by the `onSubmit` event on the form.
```suggestion
      <button type="submit" className="btn-submit">
```